test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java

```diff
@@ -456,6 +462,293 @@ public void checkpointWriterInterleavedTest() throws Exception {
         }
     }
 
+    /** Test the CheckpointWriter write size limit.
+     *
+     * CheckpointWriter aggregates a batch of SMREntries into one
+     * CheckpointEntry. This test uses large-sized SMREntries to verify
+     * that batching will not violate the maxWriteSize limit.
+     */
+    @Test
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void checkpointWriterSizeLimitTest() throws Exception {
+        final String streamName = "mystream5";
+        final UUID streamId = CorfuRuntime.getStreamID(streamName);
+        final String keyPrefix = "a-prefix";
+        final int numKeys = 50;
+        final String author = "Me, myself, and I";
+        final String mutationSuffix = "_mutation_suffix";
+        // max write size set to 25MB
+        final int maxWriteSize = 25 << 20;
+
+        StreamingMap<String, String> m = instantiateStringMap(streamName);
+
+        for (int i = 0; i < numKeys; i++) {
+            // each entry is 1 MB
+            String payload = getRandomStringOfSize(1 << 20);
+            m.put(keyPrefix + i, payload);
+        }
+
+        // set the max write size to 25 MB
+        getRuntime().getParameters().setMaxWriteSize(maxWriteSize);
+        CheckpointWriter cpw = new CheckpointWriter(getRuntime(), streamId, author, (CorfuTable) m);
+        cpw.setSerializer(serializer);
+        // set a mutator to later verify that checkpointWriter has processed the entries
+        cpw.setValueMutator((l) -> (String) l + mutationSuffix);
+
+        // Write all CP data.
+        r.getObjectsView().TXBuild()
+                .type(TransactionType.SNAPSHOT)
+                .build()
+                .begin();
+        Token snapshot = TransactionalContext
+                .getCurrentContext()
+                .getSnapshotTimestamp();
+        try {
+            cpw.startCheckpoint(snapshot);
+            cpw.appendObjectState(m.entryStream());
+            cpw.finishCheckpoint();
+
+            // Instantiate new runtime & map.
+            setRuntime();
+            Map<String, String> m2 = instantiateStringMap(streamName);
+            for (int i = 0; i < numKeys; i++) {
+                assertThat(m2.get(keyPrefix + i)).describedAs("get " + i)
+                        .isEqualTo(m.get(keyPrefix + i) + mutationSuffix);
+            }
+        } finally {
```

Comment:
This test is incorrect. `instantiateStringMap` will return the same reference for m2 and m

Reply:
Probably need to add a trim after the checkpoint too.

Reply:
You are right. Comparing `m` and `m2` is not meaningful. I changed to compare `m2` with a java hashmap object which holds the entries we originally put into `m`. As for trimming after checkpoint, following how other tests in this test suite are written (e.g. checkpointWriterTest) I didn't trim log here. I think these tests are purely for checkpointing.

Reply:
Thanks @Lujie1996

Reply:
I think you need to compare the hashmap with a new instance of CorfuTable. The CorfuTable refered by `m2` is the same as `m`

Reply:
Hi Maithem. `m2` is instantiated using a new corfu runtime (https://github.com/CorfuDB/CorfuDB/pull/2875/files#diff-bd6de4f220c80c5a84d6c60f308074aac0a096efcefe915f124f15a932652cabR514). Therefore, `m2` should be a new instance of CorfuTable, right?

Reply:
Didn't notice that `setRuntime();` switches the references. I think you're right it is a new map.